SurveyAppTests/Tests/Modules/Login/LoginPresenterSpec.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  LoginPresenterSpec.swift
+//  SurveyApp Tests
+//
+//  Created by Chananchida F. on 6/17/21.
+//
+
+import Quick
+import Nimble
+
+@testable import SurveyApp
+
+final class LoginPresenterSpec: QuickSpec {
+
+    override func spec() {
+        var router: LoginRouterInputMock!
+        var interactor: LoginInteractorInputMock!
+        var view: LoginViewInputMock!
+        var output: LoginOutputMock!
+        var presenter: LoginPresenter!
+
+        describe("a LoginPresenter") {
+            beforeEach {
+                router = LoginRouterInputMock()
+                interactor = LoginInteractorInputMock()
+                output = LoginOutputMock()
+                view = LoginViewInputMock()
+                presenter = LoginPresenter(router: router, interactor: interactor)
+
+                presenter.output = output
+                presenter.view = view
+
+            }
+
+            describe("its didPressLogin is called") {
+                context("when the parameters are valid email and password") {
+                    beforeEach {
+                        let email = "email@example.com"
+                        let password = "password"
+                        presenter.didPressLogin(email: email, password: password)
+                    }
+
+                    it("triggers interactor to call authenticateEmail") {
+                        expect(interactor.authenticateWithEmailEmailPasswordCalled) == true
+                    }
+                }
+
+                context("when the parameters are empty email and password") {
+                    beforeEach {
+                        let email = ""
+                        let password = ""
+                        presenter.didPressLogin(email: email, password: password)
+                    }
+
+                    it("should not trigger interactor to call authenticateEmail") {
+                        expect(interactor.authenticateWithEmailEmailPasswordCalled) == false
+                    }
+                }
+
+                context("when the parameter is invalid email") {
+                    beforeEach {
+                        let email = "email"
+                        let password = "password"
+                        presenter.didPressLogin(email: email, password: password)
+                    }
+
+                    it("triggers view to call showError") {
+                        expect(view.showErrorMessageCalled) == true
+                    }
+
+                    it("view should receive error message correctly") {
+                        expect(view.showErrorMessageReceivedMessage) == Localize.errorInvalidEmail()
+                    }
+                }
+            }
+
+            context("when didAuthenticateEmail is called with valid email and password") {
```

Comment:
Please update this and below one also 🙏

Reply:
@markgravity Do you mean to wrap this inside `describe` like this?
```Swift
describe("its didAuthenticateEmail is called") {
   context("when the parameter are valid email and password")
   ...
}
```